controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteScopeTask.java

```diff
@@ -0,0 +1,142 @@
+        Preconditions.checkNotNull(streamMetadataTasks);
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetadataTasks;
+        this.kvtMetadataStore = kvtMetadataStore;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteScopeEvent request) {
+        String scope = request.getScope();
+        long requestId = request.getRequestId();
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        log.debug(requestId, "Deleting {} scope recursively", scope);
+        return streamMetadataStore.checkScopeInDeletingTable(scope, context, executor).thenCompose( exists -> {
+            if (exists) {
+                return deleteScopeContent(scope, context, requestId);
+            }
+            log.info(requestId, "Skipping processing delete scope recursive for scope {} as scope" +
+                            " does not exist in deleting table", scope);
+            return CompletableFuture.completedFuture(null);
+        });
+    }
+
+    public CompletableFuture<Void> deleteScopeContent(String scopeName, OperationContext context, long requestId) {
+        List<String> readerGroupList = new ArrayList<>();
+        Iterator<Stream> iterator = listStreams(scopeName, context).asIterator();
+
+        // Seal and delete streams and add entry to RGList
+        while (iterator.hasNext()) {
+            Stream stream = iterator.next();
+            if (stream.getStreamName().startsWith(READER_GROUP_STREAM_PREFIX)) {
+                readerGroupList.add(stream.getStreamName().substring(
+                        READER_GROUP_STREAM_PREFIX.length()));
+            }
+            Futures.getThrowingException(streamMetadataTasks.sealStream(scopeName, stream.getStreamName(), requestId)
+                        .thenCompose(sealed -> streamMetadataTasks.deleteStream(stream.getScope(), stream.getStreamName(), requestId)));
+        }
+
+        // Delete ReaderGroups
+        for (String rgName: readerGroupList) {
+            Futures.getThrowingException(streamMetadataTasks.getReaderGroupConfig(scopeName, rgName, requestId)
+                    .thenCompose(conf -> streamMetadataTasks.deleteReaderGroup(scopeName, rgName,
+                            conf.getConfig().getReaderGroupId(), requestId)));
+        }
+        // Delete KVTs
+        Iterator<KeyValueTableInfo> kvtIterator = listKVTs(scopeName, requestId, context).asIterator();
+         while (kvtIterator.hasNext()) {
+         KeyValueTableInfo kvt = kvtIterator.next();
+         Futures.getThrowingException(kvtMetadataStore
+                 .deleteKeyValueTable(scopeName, kvt.getKeyValueTableName(), context, executor));
+         }
+        streamMetadataStore.deleteScopeRecursive(scopeName, context, executor)
```

Comment:
@shshashwat we have a dangling future here. You could remove the next line and on this one just do:
`return streamMetadataStore.deleteScopeRecursive(scopeName, context, executor)`

Reply:
Changed as per suggestion